### PR TITLE
fix(events): fix ftrace_hook

### PR DESCRIPTION
**Pull request by OriGlassman**
### 1. Explain what the PR does

We need to count how many ftrace based hooks will be placed on each symbol. eventsState may contain duplicate events due to dependencies. To get the real count, we consider the program name and the prob type. Furthermore, added logic that addresses the situation where there may be multiple k[ret]probes from a single probe request (due to multiple symbols at different locations).

### 2. Explain how to test it

./tracee -e=ftrace_hook

### CODE REVIEW

**Comment by geyslan** [comment]
@OriGlassman I have this PR https://github.com/aquasecurity/tracee/pull/3848 related to Events States. I think it's worth taking a look just to catch on upcoming changes.

File: pkg/ebpf/tracee.go [hunk 1]
```diff
@@ -1576,6 +1592,18 @@ func (t *Tracee) getSelfLoadedPrograms(kprobesOnly bool) map[string]int {
...
+				}
+
+				_, found := uniqueHooksMap[key]
+				if found { // Already counted this hook
+					continue
+				}
+
+				uniqueHooksMap[key] = struct{}{}
+
 				log(definition.GetName(), p.GetProgramName())
```
Comment (AlonZivony) [inline pkg/ebpf/tracee.go:1607]: I know this is not part of the current code, but shouldn't we use the `logger.Infow` here?

**Review by AlonZivony — APPROVED** [review]
LGTM
Had only one comment but its out of the PR scope

File: pkg/ebpf/tracee.go (hunk 1 above)
Comment (OriGlassman) [reply pkg/ebpf/tracee.go:1607]: This is a change done by a previous maintainer so not sure the motivation

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**